python/tests/scripts/b5_types_prelim.metta

```diff
@@ -13,97 +16,148 @@
   (Add S Z)
   S)
 
-; Adding type definitions. Arrow is a built-in symbol of the type system
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Adding type definitions.
+; Arrow (->) is a built-in symbol of the type system
 ; `Nat` is a custom symbol
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 (: Z Nat)
 (: S (-> Nat Nat))
 (: Add (-> Nat Nat Nat))
 ; Now the result of its evaluation will be a error expression, because `S` is not `Nat`
 !(assertEqualToResult
   (Add S Z)
   ((Error S BadType)))
-; This will still work, because `Something` is of undefined type,
-; and it might be `Nat`
-!(assertEqual
-  (Add Something Z)
-  Something)
-; This should work normally
+; But these other examples works normally
+!(assertEqualToResult
```

Comment:
Why is it changed to `assertEqualToResult`?

Reply:
I moved the example with "AssertEqual" below, and added a new example here

Reply:
`AssertEqualToResult` doesn't reduce the second argument, so it is used, when we want to make sure that the result is not evaluated further. For example, it is senseless to write `(AssertEqual (f x) (f x))`, because both `(f x)` will or will not be reduced. However, if we use `AssertEqualToResult` in this case, we show that `(f x)` is not reduced. Another example is when evaluation produces a error. In your case, the most important part to check is that `Add` works, and `(Add (S Z) Z)` is equivalent to `(S Z)`. Thus, using `AsserEqual` is perfectly ok. It is not a mistake to use `AssertEqualToResult`, but it is just not necessary.

Reply:
Good piece of information about `AssertEqualToResult` and `AssertEqual`, which I didn't internalize before.

The explanation for the former is in the `a1_sumbols.metta` file, but the one for `AssertEqual` comes until file `b1_equal_chain.meeta`. It could be useful to have both definitions together for contrast, perhaps in `a1`, what do you think?

Reply:
It could be useful, but it can be done in a separate PR.